packages/flutter/test/material/data_table_test.dart

```diff
@@ -866,4 +866,89 @@ void main() {
       _customHorizontalMargin,
     );
   });
+
+  testWidgets('DataTable set border width test', (WidgetTester tester) async {
+    Widget buildTable({ double thickness}) {
```

Comment:
The table built by this function seems needlessly complex for the sake of checking the divider.

It might be simpler to just create columns and rows (configured as simply as possible) and then create the DataTable's inline. Something like:

```dart
const List<DataColumn> columns = <DataColumn>[
  DataColumn(label: Text('column1')),
  DataColumn(label: Text('column2')),  
];

const List<DataCell> cells = <DataCell>[
  DataCell(Text('cell1')),
  DataCell(Text('cell2')),  
];

const List<DataRow> rows = <DataRow>[
  DataRow(cells: cells),
  DataRow(cells: cells),  
];
```

And then
```dart
await tester.pumpWidget(
  MaterialApp(
    home: Material(
      child: DataTable(
        rows: rows,
        columns, columns,
      )
    ),
  ),
);
```
etc.